test/liquid/combinators/tags/assign_test.exs

```diff
@@ -1,14 +1,77 @@
-defmodule Liquid.Combinator.Tags.AssignTest do
+defmodule Liquid.Combinators.Tags.AssignTest do
   use ExUnit.Case
 
   import Liquid.Helpers
   alias Liquid.NimbleParser, as: Parser
 
   test "assign" do
-    test_combinator(
+    tags = [
```

Comment:
This case doesn't test `assign` with string values, only integers and other variable assignments. What I'm wondering is how are we going to make the distinction between assignment with a string variable and another variable? In other words, how will we know the difference between 
`{:assign, [variable_name: "cart", value: "word"]}` and
`{:assign, [variable_name: "cart", value: "variable_word"]}`

Reply:
The string value returns "\"word\"" or "'word'" while variable returns "variable_word"

Reply:
I don't think this is the best approach. By purely looking at the AST, you have to infer that `"just_a_string"` is a variable instead of a string, which is confusing. It also doesn't make sense to parse strings containing there quotes, because later in the rendering we would have to remove these. The best approach in my opinion is to parse variables as tagged tuples. So:
```
{:assign, [variable_name: "cart", value: {:variable, "variable_name.variable_name"}]}
and strings as:
{:assign, [variable_name: "cart", value: "string"}]}
```